lib/CMakeLists.txt

```diff
@@ -1,5 +1,11 @@
 add_subdirectory(zsync2 EXCLUDE_FROM_ALL)
 
-if(NOT TARGET libappimage)
+if(NOT USE_SYSTEM_LIBAPPIMAGE)
     add_subdirectory(libappimage EXCLUDE_FROM_ALL)
+    set(APPIMAGE_LIB libappimage_shared PARENT_SCOPE)
+else()
+    find_library(APPIMAGE_LIB appimage)
```

Comment:
You should be able to find libappimage using `find_package`, since it generates an exported targets config.